### PR TITLE
reimplement as async with httpx, add version

### PR DESCRIPTION
**Pull request by faceprint**
this uses the httpx library, and lowers the default interval to 5
seconds, which has elimited the hangs and issues for me

also sets up for potentially moving ModernFormsDevice out into its own
library, which would be necessary to submit the rest to home-assistant
for inclusion

### CODE REVIEW

**Comment by icemarkom** [comment]
No comment on the rest, but I would strongly advise against dropping the scan_interval to anything under 1 minute. That is known to cause issues with a large number of MF fans.

File: custom_components/modernforms/__init__.py
Comment (icemarkom) [inline custom_components/modernforms/__init__.py]: Please see previous commit. Can you please leave this to be just a value.

File: custom_components/modernforms/__init__.py
Comment (icemarkom) [inline custom_components/modernforms/__init__.py]: Please, move the timedelta here.

File: custom_components/modernforms/__init__.py
Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: Also, I still advise we keep this at 60 seconds at a minimum.

File: custom_components/modernforms/__init__.py
Comment (icemarkom) [inline custom_components/modernforms/__init__.py:73]: Not all fans have lights. Why are you dropping this?

Comment (faceprint) [reply custom_components/modernforms/__init__.py:73]: has_light was unused outside `async_setup_entry` which should still skip the light entity if set to false

File: custom_components/modernforms/__init__.py
Comment (faceprint) [reply custom_components/modernforms/__init__.py]: Happy to move the timedelta.  I also had issues with a shorter interval before these changes, but none since.  I'd love if some others who had issues would try this and report back if they're still seeing issues.

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: I am not sure this would actually solve it. In the past, as part of the troubleshooting from HA forum, I built a completely [standalone prober](https://github.com/icemarkom/mffprober) for the fans, and simply running this for under a day would freeze them such they required a powercycle. What your solution **_may_** help with is Lovelace locking up when fans are slow to respond and/or die. My concern is an apparent memory leak on the fans themselves. Hence recommendation to leave this as a default to 60, **and** have a configuration parameter so those who need/want faster polling can adjust (see Issue #8, and potentially Issue #15).

File: custom_components/modernforms/__init__.py
Comment (icemarkom) [inline custom_components/modernforms/__init__.py]: So, to be practical, as this still concerns me. Let's roll this back to 60, and focus on other changes in this PR, because they seem to address two slightly different issues observed with the fans.

@jimpastos is, obviously, final decision maker on this one.

File: custom_components/modernforms/__init__.py
Comment (jimpastos) [reply custom_components/modernforms/__init__.py]: I would think not all fans have the same issue. My ones (AXIS) never need restarting. 
One is on firmware 01.03.0021/01.03.3014 and the other on 01.03.0021/01.03.3008. I think the latest is 0027 but I am not sure if I should attempt to update them.

@faceprint, do your fans need restarting before your fix? Either way, 5 seconds is WAY to frequent. I would think 30s or 60 is better, especially for a default value

File: custom_components/modernforms/__init__.py
Comment (jimpastos) [reply custom_components/modernforms/__init__.py]: I agree 30s or 60s is ideal. Anything lower than that should be configurable explicitly by users

File: custom_components/modernforms/__init__.py
Comment (faceprint) [reply custom_components/modernforms/__init__.py]: Yes, my fans were hanging requiring a power cycle with the "stock" integration about a week ago.  I noticed all the socket connections stuck in `TIME_WAIT` state (I think I saw 64), figuring something wasn't getting cleaned up.  I had the interval set to 10 seconds, and all was well, usually having 4 or 5 connections, all but one in `TIME_WAIT`.  I decided to give 5 seconds a try, and no longer have any connections stuck in `TIME_WAIT`  and the fan is operating normally.  This is a recent discovery, so it hasn't had time to "soak" but I'm optimistic based on the behavior of the connections.  Seems like the fan "likes" seeing activity on the connection.

As long as its configurable, I'm happy with whatever default people want.

Comment (jimpastos) [reply custom_components/modernforms/__init__.py]: Interesting. What firmware are you on?
If you set the interval to 5 seconds (without using httpx) does your theory still stand?

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: Ohhh this is indeed an interesting find! So it may not be the polling interval, but the number of sockets that kills the fan. You know what - this is something that makes me optimistic. 

A good way to test this hypothesis would be to run that prober of mine until it kills the fan, and then see if there are any open sockets. I am almost willing to bet we'll see a nice "round" number like 64.

Then, we can try the same using HTTP/1.1 with a single persistent session. This has some drawbacks, but I am almost certain Home Assistant is smart enough to re-establish it should it disconnect. 

This is rock-solid clue, first in a long time.

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: Of course, a separate issue entirely is that, right now, we don't actually have a way to override scan_interval :-).

**Review by jimpastos — APPROVED** [review]

**Review by jimpastos — APPROVED** [review]

File: custom_components/modernforms/__init__.py
Comment (jimpastos) [reply custom_components/modernforms/__init__.py]: @icemarkom will you have time to investigate this at all? If so, we'll wait before we merge these changes

Comment (faceprint) [reply custom_components/modernforms/__init__.py]: I saw some instability at 5 seconds overnight.  No hard lockups, just a lot of timeouts, and they resolved themselves (this was all overnight), so it may have been some wifi hiccup.  I think we can merge as is with 60 seconds, and experiment with other intervals to find if there's a better sweet spot.

I don't think that using the current setup at 5 seconds without async/httpx is a useful experiment, since (I believe) its the hung connections, and not the interval between them that causes the hard lockups, and the "stock" integration seems to leave a lot of those hung connections.

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: I will also test it out, but I am primary oncall in my day job until Thursday, which leaves limited time to play with the cool stuff I want to play with :-). If I find some time, will likely pull and start testing this today and let it soak for a few days.

Starting off with 10-15 seconds as the timeout may be a good thing. 5 seconds strikes me as overly excessive polling for a fan, honestly. Just sounds too chatty...

Comment (faceprint) [reply custom_components/modernforms/__init__.py]: Running for a few days with 60 seconds, saw 23 `ESTABLISHED` connections, and intermittent connectivity to the fan (even just ping has high packet loss).  Power-cycling the fan has restored normal operation for the moment.  At 60 seconds I'm (typically) seeing 1 `ESTABLISHED` and 1 `TIME_WAIT` connection at any given time.  I'm gonna try a "soak" at 10 seconds and see how that fares.

Comment (jimpastos) [reply custom_components/modernforms/__init__.py]: Could this be as simple as telling the HTTP lib no to send a connection: keep-alive? forcing polls to always close the TCP connection

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: I have only now gotten to the point I can start running Nathan's HTTPX integration.

Comment (icemarkom) [reply custom_components/modernforms/__init__.py]: And I think I am seeing the same thing as Nathan mentioned. With 60 second timeout, there always appears to be one `ESTABLISHED` and one `TIME_WAIT` connection:

```
bash-5.0# netstat -an | grep 15[12]
tcp        0      0 172.19.0.2:33606        192.168.xx.152:80       TIME_WAIT
tcp        0      0 172.19.0.2:48762        192.168.xx.151:80       ESTABLISHED
tcp        0      0 172.19.0.2:34008        192.168.xx.152:80       ESTABLISHED
tcp        0      0 172.19.0.2:48360        192.168.7xx.151:80       TIME_WAIT
```

The connections seem to be closing and reopening. I suspect that over time they just get stuck. BUT, `httpx` thing seems to be working, and is probably the right direction. 

Let's move the codebase to it, and continue troubleshooting this in issue #15.

**Pull request merged**
